docs/tutorial.rst

```diff
@@ -1,4 +1,620 @@
+
+   <!DOCTYPE html>
+   <html lang="en">
+   <head>
+     <meta charset="utf-8">
+     <title>Time zone convertor</title>
+   </head>
+   <body>
+     <h1>Time zone convertor</h1>
+     <p class="info">
+       When it's <time datetime="{src_dt.value}">{src_dt.text}</time>
+       in {src_location},<br>
+       it's <time datetime="{dst_dt.value}">{dst_dt.text}</time>
+       in {dst_location}.
+     </p>
+     <p>Go to the <a href="/">home page</a>.</p>
+   </body>
+   </html>
+
+
+Static assets
+-------------
+
+As our next step, let us apply some style to our markup.
+We create a subfolder named ``static``
+in the same folder as our Python source file
+and put a file named ``custom.css`` into that folder.
+Below is an example content for the file:
+
+.. code-block:: css
+
+   body {
+     font-family: 'Roboto', 'Helvetica', 'Arial', sans-serif;
+   }
+
+   h1 {
+     font-size: 3em;
+   }
+
+   p, h1 {
+     text-align: center;
+   }
+
+   form {
+     display: flex;
+     flex-direction: column;
+     align-items: center;
+   }
+
+   input, select, button {
+     font: inherit;
+   }
+
+   label {
+     display: block;
+   }
+
+   div.timezones {
+     display: flex;
+     justify-content: space-between;
+     margin: 1rem 0;
+   }
+
+   div.timezone {
+     width: 45%;
+   }
+
+   p.info {
+     font-size: 2em;
+     line-height: 2;
+   }
+
+   time {
+     color: #ff0000;
+   }
+
+
+The changes to the application code will be quite small.
+First, we define the file system path to the folder
+that contains the static assets:
+
+.. code-block:: python
+
+   CUR_PATH = os.path.dirname(os.path.abspath(__file__))
+   STATIC_PATH = os.path.join(CUR_PATH, "static")
+
+
+And then we add a route by creating
+a :class:`StaticApplication <clastic.static.StaticApplication>`
+with the static file system path we have defined,
+and we set it as the endpoint that will handle the requests
+to any application path under ``/static``:
+
+.. code-block:: python
+   :emphasize-lines: 4, 8
+
+   # from clastic.static import StaticApplication
+
+   def create_app():
+       static_app = StaticApplication(STATIC_PATH)
+       routes = [
+           ("/", home, "home.html"),
+           ("/show", show_time, "show_time.html"),
+           ("/static", static_app),
+       ]
+       render_factory = AshesRenderFactory(CUR_PATH)
+       return Application(routes, render_factory=render_factory)
+
+
+Don't forget to add the stylesheet link to the templates:
+
+.. code-block:: html
+
+   <head>
+     <meta charset="utf-8">
+     <title>Time zone convertor</title>
+     <link rel="stylesheet" href="/static/custom.css">
+   </head>
+
+
+Working with JSON
+-----------------
+
+In the last part of the tutorial,
+we're going to display the converted time
+in the same page as the form instead of moving to a second page.
+In order to achieve this,
+we're going to use JavaScript to update the page
+with data sent to and received from the application using JSON.
```

Comment:
Oh nice, should we call this a "basic JSON API endpoint" or something? (to be buzzword-compliant)

Reply:
Sure, we wouldn't want to miss this opportunity.